flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerDriverTest.java

```diff
@@ -669,4 +650,111 @@ void prepareForTestStartTaskExecutorProcessVariousSpec(
                     (ignore1, ignore2) -> getDriver().requestResource(taskExecutorProcessSpec));
         }
     }
+
+    @Test
+    public void testGetConatainerCompletedCauseForSuccess() throws Exception {
+        ContainerStatus containerStatus =
+                ContainerStatusPBImpl.newInstance(
+                        testingContainer.getId(),
+                        ContainerState.COMPLETE,
+                        "success exit code",
+                        ContainerExitStatus.SUCCESS);
```

Comment:
If we split the `testingGetContainerCompletedCause` into two tests then we can create a factory method for the different `ContainerStatus` that we need for the two tests.